### ISSUE TITLE
`listviews_test.dart` is failing with `vm_service` 9.4.0

### ISSUE THREAD

**Issue by elliette** [dart-lang/webdev#1741]
Before `vm_service` was at `9.3.0`, and the test was passing. Upgraded to `9.4.0`:
`> vm_service 9.4.0 (was 9.3.0)`

Running the test now fails:

```
dart test test/listviews_test.dart --no-retry --chain-stack-traces
Building package executable... (5.6s)
Built test:test.
00:15 +0 -1: _flutter.listViews [E]
  Expected: {
              'views': [
                {
                  'id': '1',
                  'isolate': {
                    'type': '@Isolate',
                    'id': '1',
                    'number': '1',
                    'name': 'main()',
                    'isSystemIsolate': false,
                    'isolateGroupId': ''
                  }
                }
              ]
            }
    Actual: {
              'views': [
                {
                  'id': '1',
                  'isolate': {
                    'type': '@Isolate',
                    'id': '1',
                    'number': '1',
                    'name': 'main()',
                    'isSystemIsolate': false,
                    'isolateGroupId': null
                  }
                }
              ]
            }
     Which: at location ['views'][0]['isolate']['isolateGroupId'] is <null> instead of ''

  package:test_api               expect
  test/listviews_test.dart 38:5  main.<fn>
```

**Comment by elliette** [comment]
Looks like this might be related to https://dart-review.googlesource.com/c/sdk/+/257265? 

Adding @liamappelbe @bkonyi for context. This is the test that is failing: https://github.com/dart-lang/webdev/blob/master/dwds/test/listviews_test.dart

Thanks!

**Comment by annagrin** [comment]
I wonder if changing final `StreamController<Map<String, Object>> _requestController;` in `dwds_vm_client` 

https://github.com/dart-lang/webdev/blob/490d6a06f7ed75654bcfd5a0e8351f8a0f59df1c/dwds/lib/src/dwds_vm_client.dart#L46

and 

https://github.com/dart-lang/webdev/blob/490d6a06f7ed75654bcfd5a0e8351f8a0f59df1c/dwds/lib/src/dwds_vm_client.dart#L46

to `StreamController<Map<String, Object?>> _requestController;` would help 


I believe this would also allow us to remove extra map creation on every VM service request here, which can be saving time:

https://github.com/dart-lang/webdev/blob/490d6a06f7ed75654bcfd5a0e8351f8a0f59df1c/dwds/lib/src/dwds_vm_client.dart#L58

**Comment by annagrin** [comment]
Ignore my previous comment, the change suggested did not help. 

Found the reason for the test failure - we set the `isolateGroupId` to `null` in dwds but `IsolateRef._fromJson` converts it to an empty string during parsing: 

```
 IsolateRef._fromJson(Map<String, dynamic> json) : super._fromJson(json) {
    id = json['id'] ?? '';
    number = json['number'] ?? '';
    name = json['name'] ?? '';
    isSystemIsolate = json['isSystemIsolate'] ?? false;
    isolateGroupId = json['isolateGroupId'] ?? ''; // This line
```

@bkonyi Is this intentional? This causes the isolate we return in `getVm` to be different from the isolate returned by the `listViews` service extension. If this is intentional behavior of the vm, then we can make it consistent in dwds by setting the `isolateGroupId` to an empty string on isolate ref creation:

https://github.com/dart-lang/webdev/blob/ef52eda986e94d5382c4aa0a3174f62263081a8c/dwds/lib/src/debugging/inspector.dart#L125

and/or

https://github.com/dart-lang/webdev/blob/ef52eda986e94d5382c4aa0a3174f62263081a8c/dwds/lib/src/debugging/inspector.dart#L141
